core/include/vecmem/memory/memory_resource.hpp

```diff
@@ -38,3 +38,54 @@ using memory_resource = std::experimental::pmr::memory_resource;
 #else
 #error "C++17 LFTS V1 (P0220R1) component memory_resource not found!?!"
 #endif
+
+#if defined(VECMEM_NOT_HAVE_DEFAULT_RESOURCE)
+
+#include <atomic>
+
+#if defined(VECMEM_HAVE_PMR_MEMORY_RESOURCE)
+namespace std::pmr {
+#else
+namespace std::experimental::pmr {
+#endif
+
+namespace detail {
+class new_delete_resource_impl : public memory_resource {
+    virtual void* do_allocate(std::size_t bytes, std::size_t) override {
+        return malloc(bytes);
+    }
+
+    virtual void do_deallocate(void* p, std::size_t, std::size_t) override {
+        free(p);
+    }
+
+    virtual bool do_is_equal(
+        const memory_resource& other) const noexcept override {
+        return &other == this;
+    }
+};
+
+inline std::atomic<memory_resource*>& default_resource() noexcept {
+    static std::atomic<memory_resource*> res{new_delete_resource()};
+    return res;
+}
+
+}  // namespace detail
+
+inline memory_resource* new_delete_resource() noexcept {
+    static detail::new_delete_resource_impl res{};
+    return &res;
+}
+
+inline memory_resource* get_default_resource() noexcept {
```

Comment:
:confused: Does this really work? `libc++` does define this function.

https://github.com/llvm/llvm-project/blob/main/libcxx/include/experimental/memory_resource#L147

I know that technically this definition does not clash with it, since the actual function in `libc++` is **not** in the `std::experimental::pmr` namespace, but rather in `std::experimental::fundamentals_v1::pmr`. (It is only pulled into the `std::pmr::experimental` namespace by a `typedef`/`using` statement.)

All in all, I have a lot of worries about this implementation...

Reply:
It works on macOS for me with Clang12. What's the alternative?

Reply:
Honestly, the underlying problem is that we're using `std::experimental::pmr` at all, and begin to polyfill by doing the typedef in `memory_resource.hpp`. Once you start doing that, all bets are off. 

Overall we'd be better off not doing that, and simply not suporting libc++.

Reply:
I think the reason why this combination of symbols works is because `fundamentals_v1` is an `inline namespace`.

I changed the polyfill to put the symbols in `inline namespace fundamentals_v1`, which is more consistent, maybe.